jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -376,8 +376,15 @@ private boolean parseRequestBuffer()
     @Override
     public void onCompleted()
     {
+        // If we are fill interested, then a read is pending and we must abort
+        if (isFillInterested())
+        {
+            LOG.warn("Pending read in onCompleted {} {}", this, getEndPoint());
+            abort(new IllegalStateException());
```

Comment:
Add a simple message to the exception. I would also change the exception to `IOException` like it is done below if we can't consume all the content.

Reply:
It is an ISE because it is a programming error.  It's really nothing to with any failure of IO, it is that we are completing a request cycle in the wrong state. 

But I'll add a message.